pkg/platform/model/buildplanner.go

```diff
@@ -637,7 +637,16 @@ func (bp *BuildPlanner) BuildTarget(owner, project, commitID, target string) err
 	return nil
 }
 
+type ErrFailedArtifacts struct {
+	Artifacts []*bpModel.Artifact
+}
+
+func (e ErrFailedArtifacts) Error() string {
+	return "ErrFailedArtifacts"
+}
```

Comment:
nit: Is this just a placeholder for the error interface? Regardless, this function could provide some value by doing some of the processing you are doing in the rationalize function. Similar to how the planning error works.

Reply:
It's just to satisfy the `error` interface. This would only ever show up in logs, as a stack trace.

I'd rather have `rationalizeError()` hold all the opinions on how these are rendered.